samtranslator/model/sam_resources.py

```diff
@@ -802,6 +803,117 @@ def _validate_deployment_preference_and_add_update_policy(
+        "FailureRetentionPeriod": PropertyType(False, is_type(int)),
+        "SuccessRetentionPeriod": PropertyType(False, is_type(int)),
+        "VpcConfig": PropertyType(False, is_type(dict)),
+        "Environment": PropertyType(False, dict_of(is_str(), is_type(dict))),
+    }
+
+    def to_cloudformation(self, **kwargs):
+        """Returns the Synthetics Canary to which this SAM Canary corresponds.
+
+        :param dict kwargs: already-converted resources that may need to be modified when converting this \
+        macro to pure CloudFormation
+        :returns: a list of vanilla CloudFormation Resources, to which this Function expands
+        :rtype: list
+        """
+        resources = []
+        synthetics_canary = self._construct_synthetics_canary()
+        resources.append(synthetics_canary)
+
+        return resources
+
+    def _construct_synthetics_canary(self):
+
+        """Constructs a AWS::Synthetics::Canary resource."""
+        canary = SyntheticsCanary(
+            self.logical_id, depends_on=self.depends_on, attributes=self.get_passthrough_resource_attributes()
+        )
+        canary.ArtifactS3Location = self.ArtifactS3Location
+        canary.Code = self._construct_code_dict
+        canary.ExecutionRoleArn = self.Role
+        canary.FailureRetentionPeriod = self.FailureRetentionPeriod
+        # sets the default name as the logical id because Synthetics Canary resource requires Name property,
+        # also requires it be lower case
+        canary.Name = self.FunctionName if self.FunctionName else self.logical_id.lower()
+        canary.RuntimeVersion = self.Runtime
+        canary.Schedule = self.Schedule
+        canary.StartCanaryAfterCreation = self.StartCanaryAfterCreation
+        canary.SuccessRetentionPeriod = self.SuccessRetentionPeriod
+        canary.Tags = self._construct_tag_list(self.Tags)
+        canary.VPCConfig = self.VpcConfig
+
+        if self.Tracing or self.Environment or self.MemorySize or self.Timeout:
+            canary.RunConfig = self._construct_run_config()
+        return canary
+
+    def _extract_not_none_properties(self, dict):
+        """
+        Filters out not None properties
+        """
+        return {k: v for k, v in dict if v is not None}
+
+    def _construct_run_config(self):
+        """
+        If the user specifies any of Tracing, MemorySize, TImeout or Environement then the RunConfig resource in the
+        transformed AWS::Synthetics::Canary needs to be added. Note, for Environment property the syntaxt in AWS::Serverless::Canary is
+        Environment:
+            Variables:
+                Var1: Var2
+        while in AWS::Synthetics::Canary its
+        EnvironmentVariables:
+            Var1: Var2
+        so it needs to be transformed accordingly
+        """
+        runconfig = {"ActiveTracing": self.Tracing, "MemoryInMB": self.MemorySize, "TimeoutInSeconds": self.Timeout}
```

Comment:
Should we use names consistent with other serverless resources? e.g. [`AWS::Serverless::Function`](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-function.html)

Reply:
I believe this is spec from https://docs.aws.amazon.com/AmazonSynthetics/latest/APIReference/API_CanaryRunConfigOutput.html